File: python/alphaloops/resources/fleet.py
"""Truck and trailer fleet data."""


class FleetResource:
    def __init__(self, http):
        self._http = http

    def trucks(self, dot_number, limit=50, offset=0):
        """Registered truck (power unit) fleet.

        Args:
            dot_number: The carrier's USDOT number.
            limit: Results per page (default 50, max 200).
            offset: Number of results to skip (default 0).
        """
        params = {"limit": limit, "offset": offset}
        return self._http.get(f"/v1/carriers/{dot_number}/trucks", params=params)

    def trucks_iter(self, dot_number, limit=200):
        """Iterate all trucks, auto-paginating."""
        offset = 0
        while True:
            resp = self.trucks(dot_number, limit=limit, offset=offset)
            trucks = resp.get("trucks", [])
            if not trucks:
                return
            yield from trucks
            offset += len(trucks)
            if offset >= resp.get("total_results", resp.get("total_trucks", 0)):
                return

    def trailers(self, dot_number, limit=50, offset=0):
        """Registered trailer fleet.

        Args:
            dot_number: The carrier's USDOT number.
            limit: Results per page (default 50, max 200).
            offset: Number of results to skip (default 0).
        """
        params = {"limit": limit, "offset": offset}
        return self._http.get(f"/v1/carriers/{dot_number}/trailers", params=params)

    def trailers_iter(self, dot_number, limit=200):
        """Iterate all trailers, auto-paginating."""
        offset = 0
        while True:
            resp = self.trailers(dot_number, limit=limit, offset=offset)
            trailers = resp.get("trailers", [])
            if not trailers:
                return
            yield from trailers
            offset += len(trailers)
            if offset >= resp.get("total_results", resp.get("total_trailers", 0)):
                return

File: typescript/src/resources/fleet.ts
import type { HTTPClient } from "../http-client.js";
import type { APIObject } from "../api-object.js";

export class FleetResource {
  constructor(private http: HTTPClient) {}

  async trucks(
    dotNumber: string,
    options: { limit?: number; offset?: number } = {}
  ): Promise<APIObject> {
    const { limit = 50, offset = 0 } = options;
    return this.http.get(`/v1/carriers/${dotNumber}/trucks`, { limit, offset });
  }

  async *trucksIter(
    dotNumber: string,
    options: { limit?: number } = {}
  ): AsyncGenerator<APIObject> {
    const { limit = 200 } = options;
    let offset = 0;
    while (true) {
      const resp = await this.trucks(dotNumber, { limit, offset });
      const trucks = resp.trucks ?? [];
      if (!trucks.length) return;
      yield* trucks;
      offset += trucks.length;
      if (offset >= (resp.total_results ?? resp.total_trucks ?? 0)) return;
    }
  }

  async trailers(
    dotNumber: string,
    options: { limit?: number; offset?: number } = {}
  ): Promise<APIObject> {
    const { limit = 50, offset = 0 } = options;
    return this.http.get(`/v1/carriers/${dotNumber}/trailers`, { limit, offset });
  }

  async *trailersIter(
    dotNumber: string,
    options: { limit?: number } = {}
  ): AsyncGenerator<APIObject> {
    const { limit = 200 } = options;
    let offset = 0;
    while (true) {
      const resp = await this.trailers(dotNumber, { limit, offset });
      const trailers = resp.trailers ?? [];
      if (!trailers.length) return;
      yield* trailers;
      offset += trailers.length;
      if (offset >= (resp.total_results ?? resp.total_trailers ?? 0)) return;
    }
  }
}
